File: app/src/main/java/com/github/gmetal/presentation/mappers/ModelMappers.java
package com.github.gmetal.presentation.mappers;

import com.github.gmetal.domain.model.MediaInfo;
import com.github.gmetal.domain.model.MovieMediaDetail;
import com.github.gmetal.presentation.model.MovieModel;

public class ModelMappers {

    public static MovieModel asMovie(final MediaInfo mediaInfo) {

        final MovieModel movieModel = new MovieModel();
        movieModel.setId(mediaInfo.getDbId());
        movieModel.setImageURL(mediaInfo.getImageURL());
        movieModel.setIsLoadingObject(false);
        movieModel.setName(mediaInfo.getName());
        movieModel.setOverview(mediaInfo.getOverview());
        movieModel.setVoteAverage(mediaInfo.getVoteAverage());
        movieModel.setVoteCount(mediaInfo.getVoteCount());
        return movieModel;
    }

    public static MovieModel asMovie(final MovieMediaDetail movieMediaDetail) {

        final MovieModel movieModel = new MovieModel();
        movieModel.setId(movieMediaDetail.getId().toString());
        movieModel.setImageURL(movieMediaDetail.getPosterPath());
        movieModel.setIsLoadingObject(false);
        movieModel.setName(movieMediaDetail.getTitle());
        movieModel.setOverview(movieMediaDetail.getOverview());
        movieModel.setVoteAverage(movieMediaDetail.getVoteAverage());
        movieModel.setVoteCount(movieMediaDetail.getVoteCount());
        return movieModel;
    }
}

File: app/src/main/java/com/github/gmetal/presentation/mappers/ModelMappers.kt
package com.github.gmetal.presentation.mappers

import com.github.gmetal.domain.model.MediaInfo
import com.github.gmetal.domain.model.MovieMediaDetail
import com.github.gmetal.domain.model.TvMediaDetail
import com.github.gmetal.presentation.model.MovieModel
import com.github.gmetal.presentation.model.TvShowModel

object ModelMappers {

    fun asMovie(mediaInfo: MediaInfo): MovieModel {

        val movieModel = MovieModel()
        movieModel.id = mediaInfo.dbId
        movieModel.imageURL = mediaInfo.imageURL
        movieModel.isLoadingObject = false
        movieModel.name = mediaInfo.name
        movieModel.overview = mediaInfo.overview
        movieModel.voteAverage = mediaInfo.voteAverage ?: 0.0f
        movieModel.voteCount = mediaInfo.voteCount ?: 0
        return movieModel
    }

    fun asMovie(movieMediaDetail: MovieMediaDetail): MovieModel {

        val movieModel = MovieModel()
        movieModel.id = movieMediaDetail.id.toString()
        movieModel.imageURL = movieMediaDetail.posterPath
        movieModel.isLoadingObject = false
        movieModel.name = movieMediaDetail.title
        movieModel.overview = movieMediaDetail.overview
        movieModel.voteAverage = movieMediaDetail.voteAverage ?: 0.0f
        movieModel.voteCount = movieMediaDetail.voteCount ?: 0
        return movieModel
    }

    fun asTvShow(tvMediaDetail: TvMediaDetail): TvShowModel {

        val movieModel = TvShowModel()
        movieModel.id = tvMediaDetail.id.toString()
        movieModel.imageURL = tvMediaDetail.posterPath
        movieModel.isLoadingObject = false
        movieModel.name = tvMediaDetail.title
        movieModel.overview = tvMediaDetail.overview
        movieModel.voteAverage = tvMediaDetail.voteAverage ?: 0.0f
        movieModel.voteCount = tvMediaDetail.voteCount ?: 0
        return movieModel
    }
}
